### DISCUSSION TITLE
Would love a right indent to be available.

### DISCUSSION THREAD

**Discussion by 4DSecurity** [PHPOffice/PhpSpreadsheet#3299] [Ideas]
The indent feature is excellent for putting a small amount of padding in cells. However, right justified text in a cell Hasn't got the same possibility which means text is always pushed hard to the right of the cell. Really could do with a **right indent** feature, as found in Excel  spreadsheet cells.

**Comment by oleibman** [comment]
The following seems to do what you want.
```php
use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\Style\Alignment;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx;

// Create new Spreadsheet object
$spreadsheet = new Spreadsheet();
$sheet = $spreadsheet->getActiveSheet();
$sheet->getCell('A1')->setValue('XYZ');
$sheet->getStyle('A1')->getAlignment()
    ->setHorizontal(Alignment::HORIZONTAL_RIGHT)
    ->setIndent(1);
$writer = new Xlsx($spreadsheet);
$outfile = 'issue.3299.indent.xlsx';
$writer->save($outfile);
```

**Reply by 4DSecurity** [reply]
Thanks for the reply, I have tried using the indent, as above, and from array. It does not appear to work for a right size padding, I can only get it to pad from the left... unless I'm missing something somewhere!

**Reply by 4DSecurity** [reply]
Hold that thought! I've just rewritten the same bit of code and it does work, you are correct. I have no idea why it wouldn't work for me before but it has to be something I did!
Can cross this idea from the list now. :)
